### PR TITLE
Trunk 4755 - Implementing Program Enrollment Attributes

### PR DESCRIPTION
**Pull request by brayoh**
## Description of what I changed
- Convert all spaces to tabs for all the changes made
- change retired attributes to voided

## Issue I worked on
[TRUNK-4755 ](https://issues.openmrs.org/browse/TRUNK-4755)

## Checklist: I completed these to help reviewers :)
- [x] My pull request only contains **ONE single commit**

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by brayoh** [comment]
@dkayiwa kindly have a look at this.

**Comment by dkayiwa** [comment]
@teleivo what do you think of this?

File: api/src/main/resources/liquibase-update-to-2.2.xml [hunk 1]
```diff
@@ -113,7 +113,7 @@
             <addForeignKeyConstraint constraintName="patient_program_attribute_changed_by_fk"
                                      baseTableName="patient_program_attribute" baseColumnNames="changed_by"
                                      referencedTableName="users" referencedColumnNames="user_id" />
-            <addForeignKeyConstraint constraintName="patient_program_attribute_retired_by_fk"
+            <addForeignKeyConstraint constraintName="patient_program_attribute_voided_by_fk"
```
Comment (teleivo) [inline api/src/main/resources/liquibase-update-to-2.2.xml]: was this intentional?

File: api/src/main/java/org/openmrs/api/ProgramWorkflowService.java [hunk 2]
```diff
@@ -43,22 +43,22 @@
  * </pre>
  */
 public interface ProgramWorkflowService extends OpenmrsService {
-	
+
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/api/ProgramWorkflowService.java]: I wouldnt add in these changes. I would only expect to see the changes you made putting in spaces reverted by changing them to tabs.

**Comment by teleivo** [comment]
@brayoh it seems you changed more than in your original commit. I would only expect the portions you changed in the commit I mentioned changed to tabs.
@dkayiwa

File: api/src/main/java/org/openmrs/api/ProgramWorkflowService.java (hunk 2 above)
Comment (brayoh) [reply api/src/main/java/org/openmrs/api/ProgramWorkflowService.java]: @teleivo I converted all the spaces I had added to tabs, hence the above changes

File: api/src/main/resources/liquibase-update-to-2.2.xml (hunk 1 above)
Comment (brayoh) [reply api/src/main/resources/liquibase-update-to-2.2.xml]: Yes it was, should I create a separate PR for this?

**Comment by brayoh** [comment]
@teleivo Let me revert what I can, I changed all files I worked with to use spaces instead of tabs, should I only change the ones you pointed out or I make changes for all the files?

File: api/src/main/java/org/openmrs/PatientProgramAttribute.java [hunk 3]
```diff
@@ -14,31 +14,31 @@
 import org.openmrs.attribute.BaseAttribute;
 
 public class PatientProgramAttribute extends BaseAttribute<ProgramAttributeType, PatientProgram> implements Attribute<ProgramAttributeType, PatientProgram> {
-    private Integer patientProgramAttributeId;
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/PatientProgramAttribute.java]: Revert these auto format changes please

File: api/src/main/java/org/openmrs/ProgramAttributeType.java [hunk 4]
```diff
@@ -33,4 +33,4 @@ public Integer getProgramAttributeTypeId() {
     public void setProgramAttributeTypeId(Integer programAttributeTypeId) {
         this.programAttributeTypeId = programAttributeTypeId;
     }
-}
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/ProgramAttributeType.java]: Revert this too

**Comment by wluyima** [comment]
Can you remove the non changes from the pull request?

**Comment by dkayiwa** [comment]
@brayoh are you still interested in finishing this up?

**Pull request closed without merging**